fix: H-13: Incorrect vault debt validation causes revert

yotov721 opened the pull request.

0xjoaovictor reviewed (commented):
This test is breaking: `test_RebalanceVaultAssetsInsufficientBalance`
<img width="1361" alt="image" src="https://github.com/user-attachments/assets/78dfd8c9-258d-4724-b4a7-c3bf3c86e7b0" />

yotov721 commented:
> This test is breaking: `test_RebalanceVaultAssetsInsufficientBalance` <img alt="image" width="1361" src="https://private-user-images.githubusercontent.com/31141391/438207017-78dfd8c9-258d-4724-b4a7-c3bf3c86e7b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4NDYyNzUsIm5iZiI6MTc0NTg0NTk3NSwicGF0aCI6Ii8zMTE0MTM5MS80MzgyMDcwMTctNzhkZmQ4YzktMjU4ZC00NzI0LWI0YTctYzNiZjNjODZlN2IwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI4VDEzMTI1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYyYTdiM2FkZTZlODVkNmNjZGE0YzY3MzllOWIxYTY4MTQ4YTIxYjNjOWI0MGIxYTAzMWNmZmI4YjQ3NDNjMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JJTZ2C7nEl9VUBlgq3vOZ0lHChi41WDNSfCs-vFxV64">

fixed, rebased and pushed
